scijava/scijava-ops-api/src/main/java/org/scijava/ops/api/features/OpMatchingException.java

```diff
@@ -1,4 +1,4 @@
-package org.scijava.ops.engine.matcher;
+package org.scijava.ops.api.features;
 
 import org.scijava.ops.api.OpRef;
 
```

Comment:
I know that `OpMatchingException` extends `RuntimeException` for downstream convenience—being able to pretend that everything is always going to be OK, and not forcing people to always catch the `OpMatchingException`—but I wonder if this is the wrong choice. From the Oracle Java documentation: "If a client can reasonably be expected to recover from an exception, make it a checked exception. If a client cannot do anything to recover from the exception, make it an unchecked exception." Is it reasonable to "recover from" a failed matching attempt? I would say yes, in the same way it's reasonable to recover from e.g. a non-existent file on the file system. Perhaps we could hedge our bets a bit, and make the matching API in `OpEnvironment` throw `OpMatchingException`, which would become checked, while `OpBuilder` could do something different, such as returning an `Optional` or null.

Reply:
scijava/scijava#85

Reply:
Resolving this, since we have the issue.